### PR TITLE
Resource limit

### PR DESCRIPTION
**Pull request by jasmingacic**
# Summary of this proposal

passing 512Mi memory limit to sync containers

## Asana task(s) link.

https://app.asana.com/0/1204316201844829/1206045382355804

## Notes for the reviewer

To spare you time on investigation I created this branch from https://github.com/calyptia/cli/pull/705

Needs https://github.com/calyptia/cloud-e2e/pull/150 for CI to pass

### CODE REVIEW

**Comment by jasmingacic** [comment]
```
	fs.StringVar(&memoryLimit, "memory-limit", "512Mi", "Minimum memory required")
```

File: k8s/client.go [hunk 1]
```diff
@@ -638,11 +639,26 @@ func (client *Client) DeployCoreOperatorSync(ctx context.Context, coreCloudURL,
...
+	resources := corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{},
+	}
+
+	ml, err := resource.ParseQuantity(memoryLimit)
+	if err == nil {
+		resources.Limits = corev1.ResourceList{
+			"memory": ml,
+		}
+	}
```
Comment (niedbalski) [inline k8s/client.go:652]: We should set a default in case this doesn't parses or comes incorrect.

**Review by niedbalski — CHANGES_REQUESTED** [review]

File: k8s/client.go (hunk 1 above)
Comment (jasmingacic) [reply k8s/client.go:652]: The default value is set to 512Mi

Comment (jasmingacic) [reply k8s/client.go:652]: We didn't have resources before. So my assumption is that it is determined on the cluster level.
I will have to look this up

**Review by niedbalski — APPROVED** [review]

**Pull request merged**